pkg/api/handlers.go

```diff
@@ -21,29 +23,39 @@ const indexPageContent = `
 		<h1>Cortex</h1>
 		<p>Admin Endpoints:</p>
 		<ul>
-			<li><a href="/config">Current Config</a></li>
-			<li><a href="/distributor/all_user_stats">Usage Statistics</a></li>
-			<li><a href="/distributor/ha_tracker">HA Tracking Status</a></li>
-			<li><a href="/multitenant_alertmanager/status">Alertmanager Status</a></li>
-			<li><a href="/ingester/ring">Ingester Ring Status</a></li>
-			<li><a href="/ruler/ring">Ruler Ring Status</a></li>
-			<li><a href="/services">Service Status</a></li>
-			<li><a href="/compactor/ring">Compactor Ring Status (experimental blocks storage)</a></li
-			<li><a href="/store-gateway/ring">Store Gateway Ring (experimental blocks storage)</a></li>
+			<li><a href="{{ .Join "/config" }}">Current Config</a></li>
+			<li><a href="{{ .Join "/distributor/all_user_stats" }}">Usage Statistics</a></li>
+			<li><a href="{{ .Join "/distributor/ha_tracker" }}">HA Tracking Status</a></li>
+			<li><a href="{{ .Join "/multitenant_alertmanager/status" }}">Alertmanager Status</a></li>
+			<li><a href="{{ .Join "/ingester/ring" }}">Ingester Ring Status</a></li>
+			<li><a href="{{ .Join "/ruler/ring" }}">Ruler Ring Status</a></li>
+			<li><a href="{{ .Join "/services" }}">Service Status</a></li>
+			<li><a href="{{ .Join "/compactor/ring" }}">Compactor Ring Status (experimental blocks storage)</a></li>
+			<li><a href="{{ .Join "/store-gateway/ring" }}">Store Gateway Ring (experimental blocks storage)</a></li>
 		</ul>
 
 		<p>Dangerous:</p>
 		<ul>
-			<li><a href="/ingester/flush">Trigger a Flush</a></li>
-			<li><a href="/ingester/shutdown">Trigger Ingester Shutdown</a></li>
+			<li><a href="{{ .Join "/ingester/flush" }}">Trigger a Flush</a></li>
+			<li><a href="{{ .Join "/ingester/shutdown" }}">Trigger Ingester Shutdown</a></li>
 		</ul>
 	</body>
-</html>`
+</html>`))
 
-func indexHandler(w http.ResponseWriter, _ *http.Request) {
-	if _, err := w.Write([]byte(indexPageContent)); err != nil {
-		http.Error(w, err.Error(), http.StatusInternalServerError)
-		return
+type templateInp struct {
+	Prefix string
+}
+
+func (i templateInp) Join(p string) string {
```

Comment:
Should we call this `PrefixedPath` or something?

Reply:
What about `AddPathPrefix`?

Reply:
Sounds good!